File: src/test/java/test/it/just/another/java/repo/NestedParenthesesTest.java
package test.it.just.another.java.repo;

import static org.junit.Assert.*;
import ie.just.another.java.repo.NestedParentheses;
import org.junit.Test;

public class NestedParenthesesTest {

    @Test
    public void checkNestedParentheses() {
        assertEquals(false,NestedParentheses.isCorrectlyNested(")("));
        assertEquals(false,NestedParentheses.isCorrectlyNested("((1)()()()()(())(())(((())))))("));
        assertEquals(true,NestedParentheses.isCorrectlyNested("(b)"));
        assertEquals(true,NestedParentheses.isCorrectlyNested(""));
    }

}

File: src/test/scala/test/ie/just/another/scala/repo/NestedParenthesesTest.scala
package test.ie.just.another.scala.repo

import ie.just.another.scala.repo.NestedParentheses
import org.scalatest.{FlatSpec, MustMatchers}

class NestedParenthesesTest extends FlatSpec with MustMatchers {

  "NestedParentheses" should "detect is parentheses are correctly nested or not" in {
    NestedParentheses.isCorrectlyNested("") mustBe true
    NestedParentheses.isCorrectlyNested("()") mustBe true
    NestedParentheses.isCorrectlyNested(")(") mustBe false

    NestedParentheses.isCorrectlyNested("((1)()((())))()") mustBe true
    NestedParentheses.isCorrectlyNested("(((a)))()()(()))(") mustBe false
  }

}
